File: GridView/EFCodeFirstInHierarchy/RadGridViewEFCodeFirstInHierarchy/RadGridViewEFCodeFirst.Data/Contracts/IRadGridViewEFCodeFirstData.cs
using System;
using System.Linq;

using RadGridViewEFCodeFirst.Models;

namespace RadGridViewEFCodeFirst.Data.Contracts
{
    public interface IRadGridViewEFCodeFirstData
    {
        IGenericRepository<Order> Orders { get; }

        IGenericRepository<OrderType> OrderTypes { get; }

        IGenericRepository<Shipper> Shippers { get; }

        void SaveChanges();
    }
}

File: GridView/EFCodeFirstInHierarchy/RadGridViewEFCodeFirstInHierarchyVB/RadGridViewEFCodeFirst.Data/Contracts/IRadGridViewEFCodeFirstData.vb
Imports RadGridViewEFCodeFirst.Models

Public Interface IRadGridViewEFCodeFirstData
    ReadOnly Property Orders() As IGenericRepository(Of Order)

    ReadOnly Property OrderTypes() As IGenericRepository(Of OrderType)

    ReadOnly Property Shippers() As IGenericRepository(Of Shipper)

    Sub SaveChanges()
End Interface

